libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -147,19 +172,17 @@ void AP_Scheduler::run(uint32_t time_available)
     if (_debug > 1 && _perf_counters == nullptr) {
         _perf_counters = new AP_HAL::Util::perf_counter_t[_num_tasks];
         if (_perf_counters != nullptr) {
-            for (uint8_t i=0; i<_num_unshared_tasks; i++) {
-                _perf_counters[i] = hal.util->perf_alloc(AP_HAL::Util::PC_ELAPSED, _tasks[i].name);
-            }
-            for (uint8_t i=_num_unshared_tasks; i<_num_tasks; i++) {
-                _perf_counters[i] = hal.util->perf_alloc(AP_HAL::Util::PC_ELAPSED, _common_tasks[i].name);
+            for (uint8_t i=_num_tasks; i<_num_tasks; i++) {
+                _perf_counters[i] = hal.util->perf_alloc(AP_HAL::Util::PC_ELAPSED, _tasks[i].task->name);
             }
         }
     }
     
     for (uint8_t i=0; i<_num_tasks; i++) {
-        const AP_Scheduler::Task& task = (i < _num_unshared_tasks) ? _tasks[i] : _common_tasks[i - _num_unshared_tasks];
+        SchedTaskListEntry &tasklist_entry = _tasks[i];
+        const AP_Scheduler::Task task = *(tasklist_entry.task);
```

Comment:
Don't you want a reference here?

Reply:
This was fixed - thanks.

OTOH, it's now dead and buried :-)